js/components/portal-dashboard/student-names.tsx

```diff
@@ -21,12 +25,22 @@ export class StudentNames extends React.PureComponent<IProps> {
           return (
             <div className={`${css.studentName} ${compactClass}`} key={`student ${i}`}>
               <div key={`student ${i}`}>
-                <div className={css.name} data-cy="student-name">{formattedName}</div>
+                <div className={css.name}
+                     data-cy="student-name"
+                     onClick={this.handleOpenStudentView("ResponseDetails", student.get("id"))}>
+                      {formattedName}
+                </div>
               </div>
             </div>
           );
         }) }
       </div>
     );
   }
+
+  private handleOpenStudentView = (mode: DashboardViewMode, currentStudentId: string) => () => {
+    this.props.setDashboardViewMode(mode);
+    this.props.setListViewMode("Question");
+    this.props.setCurrentStudent(currentStudentId);
+  }
```

Comment:
Is this function named correctly?  We call handleOpenStudentView and set the list view mode to question.  That seems backwards.

Reply:
Ah, I see the issue, we click on a student name and open the question view mode for that student.  I think this is a remnant of the previous naming.  We're really opening responseDetails in Question mode.  I would consider calling this something like handleOpenResponseDetailsInQuestionMode.  Or something that is more clear about what it does.

Reply:
It's that way because ListViewMode "Student" lists the studens and their answer to the one question. Versus ListViewMode "Question" lists the questions and their answers from the activitiy for the one particular student. The icons also show the same thing, person outline for the "Student", "Q" for the "Question"